framework/include/materials/Material.h

```diff
@@ -131,6 +131,42 @@ class Material : public MaterialBase, public Coupleable, public MaterialProperty
   const MaterialProperty<T> & getMaterialPropertyOlderByName(const std::string & prop_name);
   ///@}
 
+  /**
+   * Retrieve the dicrete material with a given parameter key named "name"
+   */
+  MaterialBase & getMaterial(const std::string & name)
+  {
+    return getMaterialByName(parameters().get<MaterialName>(name));
+  }
+
+  /**
+   * Retrieve the dicrete material named "name".
+   *
+   * @param no_warn If true, suppress warning about retrieving the material potentially during its
+   * calculation. If you don't know what this is/means, then you don't need it.
+   * @param no_dep Use no_dep = true if no dependency resolution for the material is required. Using
+   * no_dep = false is useful for discrete materials.
+   */
+  MaterialBase &
+  getMaterialByName(const std::string & name, bool no_warn = false, bool no_dep = false)
```

Comment:
this should not be in the header

Reply:
what shouldn't? The default arguments? Those should _only_ be in the header.

Reply:
The whole method was in the header: https://github.com/idaholab/moose/pull/20106/commits/f91cdb5baf10a19637ef5fd92a43f567823caa4b